libnetdata/os.h

```diff
@@ -48,8 +48,9 @@ int getsysctl_by_name(const char *name, void *ptr, size_t len);
 
 extern const char *os_type;
 
-extern int processors;
-long get_system_cpus(void);
+#define get_system_cpus() get_system_cpus_with_cache(true)
+#define get_system_cpus_uncached() get_system_cpus_with_cache(false)
+long get_system_cpus_with_cache(bool cache);
```

Comment:
i would probably call it just `_get_system_cpus` to signify it is not meant to be used directly as we have both variants to be really used as API as defines above

Reply:
I agree, but `_` as a prefix is reserved for system libraries and a lot of IDEs complain about it.

So, everywhere we do this we append `_with_X`. For example you can find a lot of references of other functions appended with `with_trace`.

If we change this, I think we should do it everywhere...